src/conn/conn_api.c

```diff
@@ -1822,9 +1821,40 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
               WT_SINGLETHREAD_STRING));
     }
 
+    /*
+     * We own the database home, figure out if we're creating it. There are a few files created when
+     * initializing the database home and we could crash in-between any of them, so there's no
+     * simple test. The last thing we do during initialization is rename a turtle file into place,
+     * and there's never a database home after that point without a turtle file. If the turtle file
+     * doesn't exist, it's a create.
+     */
+    WT_ERR(__wt_turtle_exists(session, &exist));
+    conn->is_new = exist ? 0 : 1;
+
+    /*
+     * If the turtle file exists, the WiredTiger file should exist as well, unless we are salvaging.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "salvage", &cval));
+    if (is_create && !cval.val && !conn->is_new) {
```

Comment:
Is there a difference between `is_create` and `conn->is_new`?

Reply:
I believe `is_create` is whether the "create" config is on and the connection is not "readonly", and `conn->is_new` is true if the turtle file does not exist when opening the connection. Thanks for pointing this out, as I've realised now that the check for `is_create` is not necessary since we want to treat a missing or empty WiredTiger file the same way, regardless of the "create" config.